doc/cloud-init.md

```diff
@@ -31,6 +31,111 @@ For LXD instances, `vendor-data` should be used in profiles rather than the inst
 
 Cloud-config examples can be found here: https://cloudinit.readthedocs.io/en/latest/topics/examples.html
 
+# Working with cloud-init
+
+For a safe way to test, use a new profile that's copied from the default profile.
+
+```
+$ lxc profile copy default test
+```
+
+Then edit the new `test` profile. You may wish to set your `EDITOR` environment variable first.
+
+```
+$ lxc profile edit test
+```
+
+For a new LXD installation, the configuration file should be similar to
+```yaml
+config: {}
+description: Default LXD profile
+devices:
+  eth0:
+    name: eth0
+    network: lxdbr0
+    type: nic
+  root:
+    path: /
+    pool: default
+    type: disk
+```
+
+Once you've set up the `cloud-init` configuration, use `lxc launch` with `--profile <profilename>` to apply the profile to the instance.
+
+## Adding cloud-init keys to the configuration
+
+cloud-init keys require a specific syntax. You use a pipe symbol (`|`) to indicate that LXD should pipe the contents of the relevant key
```

Comment:
That is no what the `|` does in YAML. In YAML it simply means that the key content will be on following indented lines.

Reply:
Thanks. I've never actually worked with YAML. Have now gone and read the YAML spec, and will link it in the re-worked text.